rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -22,11 +25,40 @@ public JsonError resolveError(Throwable t, Method method, List<JsonNode> argumen
         if(t instanceof  RskJsonRpcRequestException) {
             error =  new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
         } else if (t instanceof InvalidFormatException) {
-            error = new JsonError(-32603, "Internal server error, probably due to invalid parameter type", null);
+            error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error, probably due to invalid parameter type", null);
+        } else if (t instanceof UnrecognizedPropertyException) {
+            error = new JsonError(
+                    JsonRpcError.INVALID_PARAMS,
+                    getExceptionMessage((UnrecognizedPropertyException) t),
+                    null);
         } else {
             logger.error("JsonRPC error when for method {} with arguments {}", method, arguments, t);
-            error = new JsonError(-32603, "Internal server error", null);
+            error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error", null);
         }
         return error;
     }
+
+    private String getExceptionMessage(UnrecognizedPropertyException ex) {
+        if (ex.getPropertyName() == null || ex.getKnownPropertyIds() == null) return "Invalid parameters";
+
+        StringBuilder stringBuilder = new StringBuilder("Unrecognized field \"");
+        stringBuilder.append(ex.getPropertyName());
+        stringBuilder.append("\" (");
+        stringBuilder.append(ex.getKnownPropertyIds().size());
+        stringBuilder.append(" known properties: [");
+
+        Iterator<Object> iterator = ex.getKnownPropertyIds().iterator();
```

Comment:
Maybe something like this to avoid iterating with the iterator?
It's just a suggestion though, perhaps this is harder to read

```java
ex.getKnownPropertyIds().stream()
    .map(s -> String.format("\"%s\"", s))
    .joining(", ")
```

Reply:
Or maybe the stringBuilder is more performant (most likely)